File: kawazoe/q2029/tyama_codeiq2029.cs
using System;
class StepUpSum{
	static void Main(){
		for(;;){
			string s=Console.ReadLine();
			if(s==null)break;
			long n=long.Parse(s);
			if(n==0)break;
			long r=0;
			int i=2;
			for(;;i++){
				long x=n-(long)i*(i-1)/2;
				if(x<i)break;
				if(x%i==0)r+=x/i;
			}
			Console.WriteLine(r);
		}
	}
}

File: kawazoe/q2029/tyama_codeiq2029.vb
module StepUpSum
	sub Main(ByVal args() as String)
		while true
			dim s as string=Console.ReadLine()
			if s=nothing
				exit while
			end if
			dim n as long=s
			if n=0
				exit while
			end if
			dim r as long=0
			dim i as integer=2
			while true
				dim x as long=n-ctype(i,long)*(i-1)\2
				if x<i
					exit while
				end if
				if x mod i=0
					r+=x\i
				end if
				i+=1
			end while
			Console.WriteLine(r)
		end while
	end sub
end module